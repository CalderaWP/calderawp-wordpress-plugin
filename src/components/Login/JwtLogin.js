import React, {Component} from 'react';
import {TextControl,Spinner} from '@wordpress/components'

export class JwtLogin extends Component {



	constructor(props){
		super(props);
		this.state = {
			userName: '',
			password: '',
			message: '',
			loading: false
		}

		this.setUserName = this.setUserName.bind(this);
		this.setPassword = this.setPassword.bind(this);
		this.onSubmit = this.onSubmit.bind(this);
		this.getTokenViaApi = this.getTokenViaApi.bind(this);
	}



	getTokenViaApi(){
		const {
			userName,
			password
		} = this.state;
		const {
			wpApiUrl,
			onTokenReceived
		} = this.props;
		fetch( '/proxy-login', {
			method: 'post',
			headers: {
				'Accept': 'application/json',
				'Content-Type': 'application/json'
			},
			body: JSON.stringify({
				username: userName,
				password,

			})
		}).then( r => r.json() )
			.then( r => {
				this.setState({message:"logged in",loading:false});
				onTokenReceived({
					token: r.token,
					displayName: r.user_display_name
				})
			})
			.catch( e => {
				if( e.hasOwnProperty('message') ){
					console.log(e);
					this.setState({message:e.message,loading:false});
				}
			})
	}

	setUserName(userName){
		this.setState({userName});
	}

	setPassword(password){
		this.setState({password});
	}

	onSubmit(event){
		event.preventDefault();
		this.setState({loading: true});
		this.getTokenViaApi();
	}

	render(){
		const {
			userName,
			password,
			loading,
			message
		} = this.state;

		return (
			<form
				id={'caldera-pro-jwt-login'}
				onSubmit={this.onSubmit}
			>
				<div>
					{loading ? (
						<div>Loading...
						<Spinner/>
						</div>

					) : (
						<div className={'alert'}>{message}</div>
					)}
				</div>

			<TextControl
				value={userName}
				label={'User name'}
				onChange={this.setUserName}
				id={'caldera-pro-cfp-username'}
				disabled={loading}
			/>
			<TextControl
				value={password}
				type={'password'}
				label={'Password'}
				onChange={this.setPassword}
				id={'caldera-pro-cfp-password'}
				disabled={loading}
			/>
			<input type={'submit'} value={'Login'} />
		</form>);
	}
}
