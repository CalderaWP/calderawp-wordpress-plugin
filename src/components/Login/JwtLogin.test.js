import renderer from "react-test-renderer";
import React from "react";

import "@babel/polyfill";

import Enzyme, {shallow,mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({adapter: new Adapter()});
import {JwtLogin} from  './JwtLogin';


describe( 'JWT Login component', () => {
	test('Component matches snapshot', () => {
		expect(
			renderer.create(
				<JwtLogin

				/>
			)
		).toMatchSnapshot();
	});

	it( 'sets password', () => {
		const component = shallow(<JwtLogin/>);
		component.instance().setPassword('password' );
		expect( component.state('password' ) ).toBe( 'password') ;
	});

	it( 'sets username', () => {
		const component = shallow(<JwtLogin/>);
		component.instance().setUserName('userName' );
		expect( component.state('userName' ) ).toBe( 'userName') ;
	});

	test('Shows spinner when loading', () => {
		const component = shallow(<JwtLogin/>);
		component.setState( {loading:true})
		expect(
			renderer.create(component)
		).toMatchSnapshot();
	});
});
