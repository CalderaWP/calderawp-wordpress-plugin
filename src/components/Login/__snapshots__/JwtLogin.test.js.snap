exports[`JWT Login component UI Component matches snapshot 1`] = `
<form
  id="caldera-pro-jwt-login"
  onSubmit={[Function]}>
  <div>
    <div
      className="alert">
      
    </div>
  </div>
  <div
    className="components-base-control">
    <div
      className="components-base-control__field">
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-0">
        User name
      </label>
      <input
        aria-describedby={undefined}
        className="components-text-control__input"
        disabled={false}
        id="caldera-pro-cfp-username"
        onChange={[Function]}
        type="text"
        value="" />
    </div>
  </div>
  <div
    className="components-base-control">
    <div
      className="components-base-control__field">
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-1">
        Password
      </label>
      <input
        aria-describedby={undefined}
        className="components-text-control__input"
        disabled={false}
        id="caldera-pro-cfp-password"
        onChange={[Function]}
        type="password"
        value="" />
    </div>
  </div>
  <input
    id="caldera-pro-jwt-login-submit"
    type="submit"
    value="Login" />
</form>
`;

exports[`JWT Login component UI Shows spinner when loading 1`] = `
<form
  id="caldera-pro-jwt-login"
  onSubmit={[Function]}>
  <div>
    <div>
      Loading...
      <span
        className="components-spinner" />
    </div>
  </div>
  <div
    className="components-base-control">
    <div
      className="components-base-control__field">
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-2">
        User name
      </label>
      <input
        aria-describedby={undefined}
        className="components-text-control__input"
        disabled={true}
        id="caldera-pro-cfp-username"
        onChange={[Function]}
        type="text"
        value="" />
    </div>
  </div>
  <div
    className="components-base-control">
    <div
      className="components-base-control__field">
      <label
        className="components-base-control__label"
        htmlFor="inspector-text-control-3">
        Password
      </label>
      <input
        aria-describedby={undefined}
        className="components-text-control__input"
        disabled={true}
        id="caldera-pro-cfp-password"
        onChange={[Function]}
        type="password"
        value="" />
    </div>
  </div>
  <input
    id="caldera-pro-jwt-login-submit"
    type="submit"
    value="Login" />
</form>
`;
